src/EditorFeatures/Test/Diagnostics/DiagnosticAnalyzerServiceTests.cs

```diff
@@ -812,25 +756,23 @@ internal async Task TestDiagnosticSuppressor(bool includeAnalyzer, bool includeS
                     workspace.OpenDocument(document.Id);
                     var documentTrackingService = (TestDocumentTrackingService)workspace.Services.GetService<IDocumentTrackingService>();
                     documentTrackingService.SetActiveDocument(document.Id);
-                    await incrementalAnalyzer.AnalyzeDocumentAsync(document, bodyOpt: null, InvocationReasons.SemanticChanged, CancellationToken.None);
```

Comment:
these tests are bad on many levels.  Effectively both mutating the workspace, and then tryin gto simulate what the solution crawler woudl do.

I would be ok burning most of this to the ground.  But there's a lot of goodness in the test, sprinkled in with *waaaay* too much low level attempts to replicate internal impl details within the tests themselves.

Reply:
, > Effectively both mutating the workspace, and then tryin gto simulate what the solution crawler woudl do.

The only other alternative would be to author integration tests for each of these scenarios, which would be expensive in terms of implementation, maintenance as well execution time. I do not have a good answer as to how we fill up the test holes being opened up here for the Roslyn specific functionality that needs to be validated against the value of background analysis scope option set by the user, which I now presume would be owned by the lsp clients.